src/index.js

```diff
@@ -103,54 +91,121 @@ const getUtils = ({ utils }) => {
+  try {
+    const { server } = getServer({ serveDir: path, auditUrl: url });
+    const browserPath = await getBrowserPath();
+    const { error, results } = await new Promise((resolve) => {
+      server.listen(async () => {
+        try {
+          const results = await runLighthouse(browserPath, server.url);
+          resolve({ error: false, results });
+        } catch (error) {
+          resolve({ error });
+        } finally {
+          server.close();
+        }
+      });
+    });
+
+    if (error) {
+      return { error };
+    } else {
+      const { summary, shortSummary, errors } = formatResults({
+        results,
+        thresholds,
+      });
+
+      return {
+        summary,
+        shortSummary,
+        error: errors.length > 0 ? new Error(`\n${errors.join('\n')}`) : false,
+      };
+    }
+  } catch (error) {
+    return { error };
+  }
+};
+
+const processResults = ({ summaries, errors }) => {
+  if (errors.length > 0) {
+    return {
+      error: new Error(
+        errors
+          .map(({ path, url, error }) => {
+            if (path) {
+              return `\n${chalk.red('Error')} for directory '${chalk.magenta(
+                path,
+              )}': ${error.message}`;
+            }
+            if (url) {
+              return `\n${chalk.red('Error')} for url '${chalk.magenta(
+                url,
+              )}': ${error.message}`;
+            }
+            return `\n${error.message}`;
+          })
+          .join('\n'),
+      ),
+    };
+  } else {
+    return {
+      summary: summaries
+        .map(({ path, url, summary }) => {
+          if (path) {
+            return `Summary for directory '${chalk.magenta(path)}': ${summary}`;
+          }
+          if (url) {
+            return `Summary for url '${chalk.magenta(url)}': ${summary}`;
+          }
+          return `${summary}`;
+        })
+        .join('\n'),
+    };
+  }
+};
+
 module.exports = {
   onSuccess: async ({ constants, utils, inputs } = {}) => {
     const { failBuild, show } = getUtils({ utils });
 
     try {
-      const { serveDir, auditUrl, thresholds } = getConfiguration({
+      const { audits } = getConfiguration({
         constants,
         inputs,
       });
 
-      const { server } = getServer({ serveDir, auditUrl });
-
-      const browserPath = await getBrowserPath();
-
-      const { error, results } = await new Promise((resolve) => {
-        server.listen(async () => {
-          try {
-            const results = await runLighthouse(browserPath, server.url);
-            resolve({ error: false, results });
-          } catch (error) {
-            resolve({ error });
-          } finally {
-            server.close();
-          }
+      const errors = [];
+      const summaries = [];
+      for (const { path, url, thresholds } of audits) {
+        const { error, summary, shortSummary } = await runAudit({
+          path,
+          url,
+          thresholds,
         });
```

Comment:
May it be better to perform `runAudit()` in parallel?

```js
await Promise.all(audits.map(runAudit));
```

I hope `lighthouse` and `puppeteer` don't limit it number of instances. 😓

Reply:
Running Chrome in a docker container (Netlify build environment) can be quite intensive.
Not sure we would benefit from running and parallel and I'm worried it would make the results less consistent.